core/src/epicli/data/common/ansible/playbooks/roles/postgresql/tasks/replication-repmgr-Debian.yml

```diff
@@ -0,0 +1,174 @@
+---
+
+- name: Add user to sudoers file
+  lineinfile:
+    path: /etc/sudoers
+    line: 'postgres ALL=(ALL:ALL) NOPASSWD: /bin/systemctl start postgresql, /bin/systemctl stop postgresql, /bin/systemctl restart postgresql, /bin/systemctl reload postgresql'
+    validate: 'visudo -cf %s'
+
+- name: Turn on replica
+  replace:
+    path: /etc/postgresql/10/main/postgresql.conf
+    regexp: "^#wal_level = replica"
+    replace: "wal_level = replica"
+    backup: yes
+
+- name: Set max_wal_senders
+  replace:
+    path: /etc/postgresql/10/main/postgresql.conf
+    regexp: "^[#]?max_wal_senders\\s*=\\s*[0-9]+"
+    replace: "max_wal_senders = {{ specification.additional_components.replication.max_wal_senders | default('5') }}"
+
+- name: Add password for super user in postgresql
+  become_user: postgres
+  postgresql_user:
+    user: "{{ specification.additional_components.replication.user }}"
+    password: "{{ specification.additional_components.replication.password }}"
+    role_attr_flags: LOGIN,SUPERUSER
+
+- name: Add replication user to postgresql
+  become_user: postgres
+  postgresql_user:
+    user: "{{ specification.additional_components.replication.user }}"
+    password: "{{ specification.additional_components.replication.password }}"
+    role_attr_flags: LOGIN,REPLICATION
+
+- name: Add replication user to pg_hba.conf
+  lineinfile:
+    path: /etc/postgresql/10/main/pg_hba.conf
+    line: "host    replication     {{ specification.additional_components.replication.user }}     0.0.0.0/0            md5"
+
+- name: Create pgpass file
+  template:
+    src: pgpass-repmgr.j2
+    dest: /var/lib/postgresql/.pgpass
+    owner: postgres
+    group: postgres
+    mode: 0600
+
+- name: Set Node ID
+  set_fact:
+    node_id: "{{ (groups.postgresql.index(inventory_hostname) + 1) }}"
+
+- name: Create repmgr config file
+  template:
+    src: repmgr.conf.j2
+    dest: /etc/postgresql/10/main/repmgr.conf
+    owner: postgres
+    group: postgres
+    mode: 0600
+
+- name: Creates directory
+  file:
+    path: /dbbackup/{{ inventory_hostname }}/backup/
+    state: directory
+    owner: postgres
+    group: postgres
+
+- name: Insert postgresql configuration
+  blockinfile:
+    path: /etc/postgresql/10/main/postgresql.conf
+    marker: "# ANSIBLE MANAGED BLOCK"
+    insertafter: "# Add settings for extensions here"
+    block: |
+      archive_mode = on
+      archive_command = 'test ! -f /dbbackup/{{ inventory_hostname }}/backup/%f && gzip -c < %p > /dbbackup/{{ inventory_hostname }}/backup/%f'
+      shared_preload_libraries = 'repmgr'
+
+- name: Create a new database for repmgr
+  become_user: postgres
+  postgresql_db:
+    name: "{{ specification.additional_components.replication.repmgr_database }}"
+    owner: "{{ specification.additional_components.replication.user }}"
+
+# Master:
```

Comment:
Restarts could be done conditionally, i.e. `when: task_name.changed` and group inside Ansible block to have common when condition.